src/loaders/ObjectLoader.js

```diff
@@ -113,17 +112,19 @@ class ObjectLoader extends Loader {
 
 	}
 
-	parse( json, onLoad ) {
+	parse() {
```

Comment:
The `load()` method uses `parse()` here:

https://github.com/mrdoob/three.js/pull/22026/files#diff-e1081b3871dbc5e4ca455aee64b5b8c0fd497556bf26276ef3044f2a67f8e694R109

Reply:
Oh...

Reply:
Not sure how to solve this 🤔

Reply:
Although the editor is the most important stakeholder of `ObjectLoader`, the loader is also occasionally part of user level and example code.

I wonder if we could risk a breaking change and make both `load()` and `parse()` async. This would allow us to finally solve #13177 and make the implementation less hacky.

However, a migration task would be required which could be a complicated one for  inexperienced users. I've frequently seen users struggling with the async/await pattern and the respective integration in their sync code base.

Besides, `ObjectLoader` is derived from `Loader` which means  `load()` and `parse()` are not considered to be async. I guess it would be necessary to remove the derivation if `load()` and `parse()` would be async.

Um, maybe better to rethink this change. It feels a little bit dangerous :see_no_evil:.

Reply:
Indeed!

Maybe better to just add new `loadAsync()` and `parseAsync()`.